packages/router5/index.d.ts

```diff
@@ -56,68 +57,332 @@ declare module "router5" {
+     * 
+     * @param startPathOrState The start path or state. This is optional when using the browser plugin.
+     * @param done A done node style callback (err, state)
+     * @returns The router instance
+     */
+    start(startPathOrState ?: string|State, done?:Function): Router;
+
+    /**
+     * Stop the router
+     * 
+     * @returns The router instance
+     */
     stop(): Router;
 
-    // navigation
-    cancel(): Router;
-    forward(fromRouteName: string, toRouteName: string): Router;
-    navigate(routeName: string, routeParams?: object, options?: NavigationOptions, done?: Function): Function;
-    navigateToDefault(options?: NavigationOptions, done?: Function): Function;
-
-    // route lifecycle
-    canActivate(name: string, canActivateHandler: Function | boolean): Router;
-    canDeactivate(name: string, canDeactivateHandler: Function | boolean): Router;
-    clearCanDeactivate(name: string): Router;
-
-    // middlewares
-    useMiddleware(...middlewares: Array<MiddlewareFactory>): Router;
-    clearMiddleware(): Router;
-
-    // plugins
-    usePlugin(...plugins: Array<PluginFactory>): Router;
-    hasPlugin(name: string): boolean;
-
-    // utils
-    isActive(name: string, params: object, strictEquality?: boolean, ignoreQueryParams?: boolean): boolean;
-    areStatesEqual(state1: State, state2: State, ignoreQueryParams: boolean): boolean;
-    areStatesDescendants(parent: State, child: State): boolean;
-    buildPath(route: string, params: object): string;
-    matchPath(path: string, source?: string): object;
-    setRootPath(rootPath: string): void;
+    /**
+     * Register middleware functions.
+     * 
+     * @param args The middleware functions
+     * @returns The router instance
+     */
+    useMiddleware(...args: Array<MiddlewareFactory>): Router;
+
+    /**
+     * Use plugins
+     * @param pluginFactory An argument list of plugins
+     * @returns The router instance
+     */
+    usePlugin(pluginFactory: PluginFactory): Router;
+
+    /**
+     * Set the root node path, use carefully. It can be used to set app-wide allowed query parameters.
+     * 
+     * @param rootPath The root node path
+     */
+    setRootPath(rootPath : string) : void;
   }
 
+  /**
+   * The result can be synchronous (returning a boolean) or asynchronous (returning a promise or calling done(err, result))
+   */
+  export type RouterActivationHandler = (tostring: State, fromState: State, done:(err:any,result:any) => void) => boolean | Promise<boolean> | undefined;
+
+  export interface Route {
+    name: string;
+    path: string;
+
+    /**
+     * if specified, the router will transition to the forwarded route instead. It is useful for defaulting to a child route, or having multiple paths pointing to the same route.
+     */
+    forwardTo?: string;
+
+    /**
+     * a method to control whether or not the route node can be activated
+     */
+    canActivate?: (router: Router) => RouterActivationHandler;
```

Comment:
Can you make this `RouterActivationHandler`?

Reply:
Based on your [docs](http://router5.github.io/docs/preventing-navigation.html) the function accepts a router and returns a callback. Therefore this definition should be correct.

    const canActivate = (router) => (toState, fromState) => {
         return true;
    }

    router.canActivate('admin', canActivate);